modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaUpdateCommerceAddressCheck.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.petra.string.StringUtil;
+import com.liferay.source.formatter.check.util.SourceUtil;
+import com.liferay.source.formatter.parser.JavaClass;
+import com.liferay.source.formatter.parser.JavaClassParser;
+import com.liferay.source.formatter.parser.JavaMethod;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Michael Cavalcanti
+ */
+public class UpgradeJavaUpdateCommerceAddressCheck extends BaseUpgradeCheck {
+
+	@Override
+	protected String format(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (!childJavaTerm.isJavaMethod()) {
+				continue;
+			}
+
+			JavaMethod javaMethod = (JavaMethod)childJavaTerm;
+
+			String javaMethodContent = javaMethod.getContent();
+
+			Pattern pattern = _getPattern();
+
+			Matcher matcher = pattern.matcher(javaMethodContent);
+
+			while (matcher.find() &&
+				   _isCommerceAddress(javaMethodContent, content, matcher)) {
+
+				String line = matcher.group();
+
+				String newLine = StringUtil.replace(
+					line, matcher.group(1),
+					_getNewUpdateCommerceAddressImplementation(
+						SourceUtil.getIndent(line), matcher.group(2)));
+
+				javaMethodContent = StringUtil.replace(
+					javaMethodContent, line, newLine);
+			}
+
+			content = StringUtil.replace(
+				content, javaMethod.getContent(), javaMethodContent);
+		}
+
+		return content;
+	}
+
+	@Override
+	protected String[] getNewImports() {
+		return new String[] {
+			"com.liferay.portal.kernel.service.ServiceContextThreadLocal"
+		};
+	}
+
+	private String _getNewUpdateCommerceAddressImplementation(
+		String indent, String variable) {
+
+		StringBundler sb = new StringBundler(7);
+
+		sb.append(StringPool.OPEN_PARENTHESIS);
+
+		for (String method : _METHODS) {
+			sb.append(StringPool.NEW_LINE);
+			sb.append(indent);
+			sb.append(StringPool.TAB);
+			sb.append(variable);
+			sb.append(StringPool.PERIOD);
+			sb.append(method);
+			sb.append("(),");
+		}
+
+		sb.append(StringPool.NEW_LINE);
+		sb.append(indent);
+		sb.append(StringPool.TAB);
+		sb.append("ServiceContextThreadLocal.getServiceContext()");
+		sb.append(StringPool.NEW_LINE);
+		sb.append(indent);
+		sb.append(StringPool.CLOSE_PARENTHESIS);
+
+		return sb.toString();
+	}
+
+	private Pattern _getPattern() {
+		return Pattern.compile(
+			"\\t+\\w+\\.\\s*updateCommerceAddress(\\(\\s*(\\w+)\\s*\\));");
+	}
```

Comment:
This is not a `BaseUpgradeMatcherReplacementCheck` check, so I'm wondering why you created a method to get the regex pattern. Also, we are building the `Pattern` multiple times by doing this, which can be expensive when we are talking about regex.

Reply:
Hi @kevhlee.
I didn't use the BaseUpgradeMatcherReplacementCheck class because I need to work with each method independently so there's no possibility that a method call from one method interferes with another method's check.
I created the getPattern method because the intention in the future is to create other base classes so, in order to reduce rework, I already created it following the standards of other classes, but if this is a problem, I can go back to the original pattern.
About the pattern being created several times I agree that it can be placed outside the loop so that this does not happen.

Reply:
Hey @michaelmdelcavalcanti, I see. Regarding the `_getPattern()` method, I suggest removing it because it is unnecessary at the moment and it only slows down the check.